src/Sylius/Bundle/ApiBundle/Tests/CommandHandler/ChangeAddressOrderHandlerTest.php

```diff
@@ -0,0 +1,195 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Tests\CommandHandler;
+
+use Doctrine\Common\DataFixtures\Purger\ORMPurger;
+use Doctrine\Persistence\ObjectManager;
+use SM\Factory\FactoryInterface as StateMachineFactoryInterface;
+use Sylius\Bundle\ApiBundle\Command\Checkout\AddressOrder;
+use Sylius\Bundle\ApiBundle\CommandHandler\Checkout\AddressOrderHandler;
+use Sylius\Bundle\ApiBundle\Mapper\AddressMapperInterface;
+use Sylius\Component\Addressing\Model\AddressInterface;
+use Sylius\Component\Core\Model\Address;
+use Sylius\Component\Core\Model\Channel;
+use Sylius\Component\Core\Model\Order;
+use Sylius\Component\Core\Repository\CustomerRepositoryInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Currency\Model\Currency;
+use Sylius\Component\Locale\Model\Locale;
+use Sylius\Component\Resource\Factory\FactoryInterface;
+use Sylius\Component\Resource\Repository\RepositoryInterface;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+
+final class ChangeAddressOrderHandlerTest extends KernelTestCase
+{
+    /**
+     * @test
+     */
+    public function it_changes_address_order_without_duplication_in_database(): void
+    {
+        $container = self::bootKernel()->getContainer();
+
+        /** @var RepositoryInterface $addressRepository */
+        $addressRepository = $container->get('sylius.repository.address');
+
+        /** @var RepositoryInterface $channelRepository */
+        $channelRepository = $container->get('sylius.repository.channel');
+
+        /** @var RepositoryInterface $currencyRepository */
+        $currencyRepository = $container->get('sylius.repository.locale');
+
+        /** @var CustomerRepositoryInterface $customerRepository */
+        $customerRepository = $container->get('sylius.repository.customer');
+
+        /** @var RepositoryInterface $localeRepository */
+        $localeRepository = $container->get('sylius.repository.locale');
+
+        /** @var OrderRepositoryInterface $orderRepository */
+        $orderRepository = $container->get('sylius.repository.order');
+
+        /** @var FactoryInterface $customerFactory */
+        $customerFactory =  $container->get('sylius.factory.customer');
+
+        /** @var ObjectManager $manager */
+        $manager = $container->get('doctrine.orm.default_entity_manager');
+
+        /** @var StateMachineFactoryInterface $stateMachineFactory */
+        $stateMachineFactory = $container->get('sm.factory');
+
+        /** @var AddressMapperInterface $addressMapper */
+        $addressMapper = $container->get('Sylius\Bundle\ApiBundle\Mapper\AddressMapperInterface');
+
```

Comment:
This setup is so big 😢 Is there any chance we could use some fixtures for that? Nelmio maybe?